src/js/components/Layer/StyledLayer.js

```diff
@@ -69,15 +69,32 @@ export const StyledOverlay = styled.div`
     )} pointer-events: all;
 `;
 
-const MARGINS = {
-  top: (margin, theme) =>
-    theme.global.edgeSize[margin.top || margin.vertical || margin] || '0px',
-  bottom: (margin, theme) =>
-    theme.global.edgeSize[margin.bottom || margin.vertical || margin] || '0px',
-  left: (margin, theme) =>
-    theme.global.edgeSize[margin.left || margin.horizontal || margin] || '0px',
-  right: (margin, theme) =>
-    theme.global.edgeSize[margin.right || margin.horizontal || margin] || '0px',
+const getMargin = (margin, theme, position) => {
+  const axe =
+    position === 'top' || position === 'bottom' ? 'vertical' : 'horizontal';
+  const marginApplied =
+    theme.global.edgeSize[margin[position] || margin[axe] || margin] ||
```

Comment:
can we do 

`const marginValue = margin[position] || margin[axe] || margin;` ?

Reply:
the .  `theme.global.edgeSize[marginValue] || marginValue`;